### ISSUE TITLE
Winding error in convert_nef_polyhedron_to_polygon_mesh with triangulation

### ISSUE THREAD

**Issue by mkovacs** [CGAL/cgal#3871]
## Issue Details

`CGAL::convert_nef_polyhedron_to_polygon_mesh<Nef_polyhedron_3, Surface_mesh>` with `triangulate_all_faces = true` produces a mesh that has a triangle with the wrong winding, so when viewed, the mesh seems to have a hole. The input `Nef_polyhedron_3` is simple, and is the difference of two cubes, one rotated and translated.

## Source Code

https://github.com/mkovacs/repro-cgal-nef3-triangulation-bug/commit/e5aabaaa8c70bda60b375864040fe06dcae61272
If you have Nix, you can build via `nix-build` from the repo root.

## Environment

* Operating system: Linux 64 bit
* Compiler: GCC 7.4.0
* Release or debug mode: Release
* CGAL version: 4.14
* Boost version: 1.67.0

## Output
```
OFF
14 23 0
-1 -1 -1
-1 -1 1
1 -1 1
1 -1 -1
1 1 -1
-1 1 -1
-1 1 1
-0.129104 1 1
-0.300511 0.429614 1
1 -0.129104 1
1 -0.300511 0.429614
1 1 -0.129104
0.429614 1 -0.300511
-3.33067e-16 -3.33067e-16 -3.33067e-16
3 1 0 2
3 2 0 3
3 3 0 4
3 4 0 5
3 5 0 6
3 6 0 1
3 9 8 2
3 2 8 1
3 8 6 1
3 7 6 8
3 11 10 4
3 4 10 3
3 10 2 3
3 9 2 10
3 7 12 6
3 6 12 5
3 12 4 5
3 11 4 12
3 13 12 8
3 8 12 7
3 9 10 8
3 8 10 13
3 10 11 12
# is_simple: 1
```

**Comment by mkovacs** [comment]
Taking the `interior`, `closure`, or `regularization` of the result before conversion changes which triangles in the output have the wrong winding.

**Comment by sloriot** [comment]
Thanks for the report, #3876 should fix the issue.

**Comment by lrineau** [comment]
Fixed by #3876.

**Comment by mkovacs** [comment]
Thank you for fixing this!
